### PR TITLE
feat: modularisation

### PR DESCRIPTION
**Pull request by nugaon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,17 +38,21 @@
...
     "qrcode.react": "1.0.1",
-    "react": "17.0.2",
     "react-copy-to-clipboard": "5.0.4",
-    "react-dom": "17.0.2",
     "react-feather": "2.0.9",
     "react-identicons": "1.2.5",
     "react-router-dom": "5.2.0",
     "react-syntax-highlighter": "15.4.4",
-    "semver": "7.3.5",
+    "semver": "^7.3.5",
```
Comment (vojtechsimetka) [inline package.json:46]: could we keep the dependencies locked?

File: src/config.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    this.BEE_DOCS_HOST = getProcessEnv('REACT_APP_BEE_DOCS_HOST') || 'https://docs.ethswarm.org/docs/'
+    this.BEE_DISCORD_HOST = getProcessEnv('REACT_APP_BEE_DISCORD_HOST') || 'https://discord.gg/eKr9XPv7'
+    this.GITHUB_REPO_URL =
+      getProcessEnv('REACT_APP_BEE_GITHUB_REPO_URL') || 'https://api.github.com/repos/ethersphere/bee'
+  }
+}
+
+export const config = new Config()
+
+export default config
```
Comment (vojtechsimetka) [inline src/config.ts:29]: This can probably just be an object and not a full fledged class, right?

File: src/providers/Settings.tsx [hunk 3]
```diff
@@ -42,6 +53,17 @@ export function Provider({ children }: Props): ReactElement {
     }
   }, [apiUrl])
 
+  useEffect(() => {
+    // eslint-disable-next-line no-console
+    console.log('bee-api-url2', beeApiUrl)
```
Comment (vojtechsimetka) [inline src/providers/Settings.tsx:58]: please remove

File: webpack.config.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          },
+        },
+      ],
+    },
+    // plugins: [
+    //   new ProvidePlugin({
+    //     React: 'react',
+    //     'window.React': 'react',
+    //   }),
+    // ],
```
Comment (vojtechsimetka) [inline webpack.config.ts:65]:
```suggestion
    }
```

File: package.json [hunk 5]
```diff
@@ -36,17 +38,21 @@
     "notistack": "1.0.10",
     "opener": "1.5.2",
     "qrcode.react": "1.0.1",
-    "react": "17.0.2",
```
Comment (vojtechsimetka) [inline package.json:39]: this should stay as dependency

File: package.json [hunk 6]
```diff
@@ -36,17 +38,21 @@
...
     "react-copy-to-clipboard": "5.0.4",
-    "react-dom": "17.0.2",
```
Comment (vojtechsimetka) [inline package.json:41]: this should stay as dependency

**Review by vojtechsimetka — COMMENTED** [review]
Two general comments:
- Please lock the versions
- Do we need `babel` and `webpack`? My fear is that this will start clashing (version wise) with `react-scripts` (got that experience in past) and this pattern is heavily discouraged.

BTW: Great job cleaning up the config :)

File: src/config.ts (hunk 2 above)
Comment (nugaon) [reply src/config.ts:29]: can be. I made it like this because of the `readonly` prop.
what would be the advantage of that?

Comment (vojtechsimetka) [reply src/config.ts:29]: just better readability IMHO but let's keep it as is, you are right it's less error prone this way.

**Review by vojtechsimetka — COMMENTED** [review]
Looks good, please do lock all the dependencies if possible (especially as this is now going to be included in other projects).

**Comment by nugaon** [comment]
> Looks good, please do lock all the dependencies if possible (especially as this is now going to be included in other projects).

as we discussed on one of our roundtable, it is not really necessary if you use `npm ci` instead of `npm i`.

p.s. since then some wild conflicts appeared I'm going to make a force push after resolving those.

**Comment by vojtechsimetka** [comment]
> > Looks good, please do lock all the dependencies if possible (especially as this is now going to be included in other projects).
> 
> as we discussed on one of our roundtable, it is not really necessary if you use `npm ci` instead of `npm i`.

Can you please refer me to that conclusion/discussion? Maybe it was done in my absence. AFAIK I presented this topic on roundtable in august and there the conclusion was that at least for the UI project we use fixed version (although this is not recorded in the notes). I am happy to reopen this discussion, just wanted to follow our guidelines - https://roamresearch.com/#/app/swarm-org/page/K97knoZCm

**Comment by nugaon** [comment]
> AFAIK I presented this topic on roundtable in august and there the conclusion was that at least for the UI project we use fixed version (although this is not recorded in the notes)

as I remember it was after that when I presented in npm security topic.

Nevertheless, I upload the package.json with the fixed version numbers, because it cannot cause any bad thing it is just incovenient to do so. That's why `package-lock.json` exists.

**Review by vojtechsimetka — APPROVED** [review]
LGTM. Will probably have to rebase/resolve conflicts with master

**Comment by nugaon** [comment]
> LGTM. Will probably have to rebase/resolve conflicts with master

there are no conflicts.

**Pull request merged**